signature/policy_eval_signedby.go

```diff
@@ -47,20 +47,22 @@ func (pr *prSignedBy) isSignatureAuthorAccepted(image types.UnparsedImage, sig [
 		return sarRejected, nil, err
 	}
 	defer os.RemoveAll(dir)
-	mech, err := newGPGSigningMechanismInDirectory(dir)
-	if err != nil {
-		return sarRejected, nil, err
-	}
+	/*
+		mech, err := newGPGtypes.SigningMechanismInDirectory(dir)
+		if err != nil {
+			return sarRejected, nil, err
+		}
+	*/
```

Comment:
> the build tags are the "last resort" to me as the management of them proved cumbersome in openshift (we forgot to add it for registry that resulted in storage driver disabled in release, etc..)

This seems like a kind of thing which could be once for all ensured using a build-time test.

> The cleanest (in go way) solution would be to separate the packages and have interface for them.

Sure, but _something somewhere_ has to make a choice, and forcing callers who do not care (they may not even know that cryptography is involved somewhere in the stack) to make such a choice is not really clean either. (In the worst case, in the future, we could have a caller who provides a X.509 signature, knows that, and is still forced to make a choice between OpenPGP implementations.)

> The interface (SigningMechanism) is not really a good interface if an implementation is used instead anyway of it.

Look at #207 - the interface stays, and really is not the point; the point is what and how chooses the which implementation is used.